### PR TITLE
Upgrade TypeScript to v3.6

### LINKED ISSUE #713: TypeScript 3.6 breaks how JSDoc comments work
## source

```typescript
/** @noSelfInFile */

/**
 * This is the require docstring
 */
declare function require(modname: string): any;

require("test")
```

## compiled
```lua
require(nil, "test")
```

## expected
```lua
require("test")
```

### PR DESCRIPTION
**Pull request by ark120202**
Closes #713.

### CODE REVIEW

**Comment by Perryvw** [comment]
Do we have a test that fails in TS 3.6 because of this change? If not we shoudl probably add one, a snapshot test with `/** @noSelfInFile */` and some other JSDoc on the first statement would suffice.

File: src/TSHelper.ts [hunk 1]
```diff
@@ -419,10 +419,25 @@ export function getCustomNodeDirectives(node: ts.Node): Map<DecoratorKind, Decor
 }
 
 export function getCustomFileDirectives(file: ts.SourceFile): Map<DecoratorKind, Decorator> {
+    const directivesMap = new Map<DecoratorKind, Decorator>();
+
     if (file.statements.length > 0) {
-        return getCustomNodeDirectives(file.statements[0]);
+        // Manually collect jsDoc because `getJSDocTags` includes tags only from closest comment
+        const jsDoc = file.statements[0].jsDoc;
```
Comment (tomblind) [inline src/TSHelper.ts]: Is it possible for a file to have no statements?

Comment (ark120202) [reply src/TSHelper.ts]: Yes, for empty file `SourceFile.statements` is an empty array, that's why there is a length check above.

Comment (tomblind) [reply src/TSHelper.ts]: Sorry - I completely missed that

**Pull request merged**